cache.go

```diff
@@ -16,6 +17,15 @@ import (
 var ErrCacheMiss = errors.New("cache: key is missing")
 var errRedisLocalCacheNil = errors.New("cache: both Redis and LocalCache are nil")
 
+type Logger interface {
+	Printf(format string, v ...interface{})
+}
+
+var (
+	// Log is the instance of a Logger interface that cache writes errors to.
+	Log Logger = log.New(os.Stderr, "", log.LstdFlags)
```

Comment:
Please unexport this var and hide it behide a setter function `SetLogger`. After that we should be able to merge this.

Reply:
Do you mind if I place the `Log` variable and the `Logger` interface to the `internal` package the way you did in [go-redis/redis](https://github.com/go-redis/redis/blob/e9233d8d17b43869f175103b038f9fa642518384/internal/log.go)?

Reply:
No, that sounds good.

Reply:
Done